// Copyright 2019 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe;

import "mediapipe/framework/calculator.proto";

message TensorToVectorIntCalculatorOptions {
  extend mediapipe.CalculatorOptions {
    optional TensorToVectorIntCalculatorOptions ext = 464933130;
  }

  // If true, unpack a 2d tensor (matrix) into a vector<vector<float>>. If
  // false, convert a 1d tensor (vector) into a vector<float>.
  optional bool tensor_is_2d = 1 [default = false];

  // If true, an N-D tensor will be flattened to a vector<float>. This is
  // exclusive with tensor_is_2d.
  optional bool flatten_nd = 2 [default = false];

  // If true, represents the vector as tokens and outputs just the position
  // of values above the threshold into the output vector.
  optional bool tensor_is_token = 3 [default = false];
  // Threshold for the token generation.
  optional float token_threshold = 4 [default = 0.5];
}
